sdv/data_processing/data_processor.py

```diff
@@ -103,6 +104,112 @@ def __init__(self, metadata, enforce_rounding=True, enforce_min_max_values=True,
+        """
+        try:
+            self._check_import_address_transformers()
+            return [
+                item for col_tuple, transformer in self.columns_to_multi_col_transformer.items()
+                if isinstance(
+                    transformer, (
+                        rdt.transformers.RandomLocationGenerator,
+                        rdt.transformers.RegionalAnonymizer
+                    )
+                ) for item in col_tuple
+            ]
+        except ImportError:
+            return []
+
+    def _get_address_transformer_parameters(self, column_names):
+        """Get the parameters for the address transformer.
+
+        Args:
+            column_names (tuple[str]):
+                The column names to get the parameters for.
+
+        Returns:
+            dict:
+                A dictionary mapping column names to sdtypes.
+        """
+        columns_to_sdtypes = {}
+
+        for column in column_names:
+            sdtype = self.metadata.columns[column]['sdtype']
+            columns_to_sdtypes[column] = sdtype
+
+        return columns_to_sdtypes
+
+    def _get_address_transformer(self, anonymization_level):
+        """Get the address transformer.
+
+        Args:
+            anonymization_level (str):
+                The anonymization level for the address transformer.
+        """
+        locales = self._locales if self._locales else ['en_US']
+        self._check_import_address_transformers()
+        transformer = None
+        if anonymization_level == 'full':
+            transformer = rdt.transformers.RandomLocationGenerator(locales=locales)
+        elif anonymization_level == 'street_address':
+            transformer = rdt.transformers.RegionalAnonymizer(locales=locales)
+
+        return transformer
+
+    def _update_address_transformer(self, transformer, columns_to_sdtypes):
+        """Update the transformer parameters.
+
+        Args:
+            transformer (rdt.transformers):
+                The transformer to update.
+            columns_to_sdtypes (dict):
+                A dictionary mapping column names to sdtypes.
+        """
+        list_sdtypes = list(columns_to_sdtypes.values())
+        transformer.columns_to_sdtypes = columns_to_sdtypes
+        transformer._list_sdtypes = list_sdtypes
+
+    def _set_address_transformer(self, column_names, anonymization_level):
+        """Set the address transformer.
+
+        Args:
+            column_names (tuple[str]):
+                The column names to set the transformer for.
+            anonymization_level (str):
+                The anonymization level for the address transformer.
+        """
+        columns_to_sdtypes = self._get_address_transformer_parameters(column_names)
+        transformer = self._get_address_transformer(anonymization_level)
+        self._update_address_transformer(transformer, columns_to_sdtypes)
+        transformer._validate_sdtypes()
```

Comment:
The error messages described in the issue are slightly different than that of what the transformer returns. Perhaps we can change those errors to be class methods or static methods on the transformer that take in the column_to_sdtype dict as an argument. Then we can call without initializing the transformers here which seems a little awkward